WordPress/Classes/Utility/WPTableViewHandler.m

```diff
@@ -657,6 +659,12 @@ - (void)controller:(NSFetchedResultsController *)controller
      forChangeType:(NSFetchedResultsChangeType)type
       newIndexPath:(NSIndexPath *)newIndexPath
 {
+    // Prevent crashes in iOS 14 if the row is negative
+    // See http://git.io/JIKIB
+    if (indexPath.row < 0 || newIndexPath.row < 0) {
```

Comment:
I'm not 100% sure about this change here.

In theory, iOS 13 only ignores the operations in the tableView. The rest of the code (like clearing cached row heights) is still executed though.

I'm just afraid that preventing the execution here will lead to weird states in the tableView.

Another option would be to create an extension for the `UITableView` and do the row checking there, and use those methods here.

Reply:
> I'm just afraid that preventing the execution here will lead to weird states in the tableView

In that case I wonder if would be good to [self.tableView reloadData] before returning?

Reply:
@aerych I'm more worried about `[self clearCachedRowHeightsBelowIndexPath:newIndexPath];` not being called, for example.

_In theory_, a negative row shouldn't make any difference for the `UITableView`. I guess that some inconsistencies happen when the indexPath is positive and not present, in this case, a `reloadData` might help.

Anyway, a minor issue with the TableView (that can be solved by a pull to refresh) seems to have a smaller impact than the app crashing.

Wdyt?

Reply:
I just looked at it again and I don't think that clearing (or invalidating) the cache of a "negative row" is necessary. :)